File: code/Task.cpp
// Бинарная куча:
#include <queue>
std::priority_queue<int> maxHeap;           // максимальная куча
std::priority_queue<int, std::vector<int>, std::greater<int>> minHeap; // минимальная куча

// Операции:
maxHeap.push(value);                        // добавление элемента
maxHeap.pop();                             // удаление вершины
maxHeap.top();                             // доступ к вершине

//Хеш таблица
#include <unordered_map>
std::unordered_map<KeyType, ValueType> hashTable;

// Операции:
hashTable[key] = value;                     // вставка/обновление
hashTable.find(key);                       // поиск
hashTable.erase(key);                      // удаление
hashTable.size();                          // размер

File: code/Task.java
// Бинарная куча
import java.util.PriorityQueue;
PriorityQueue<Integer> minHeap = new PriorityQueue<>();    // минимальная куча
PriorityQueue<Integer> maxHeap = new PriorityQueue<>((a,b) -> b-a); // максимальная куча

// Операции:
minHeap.add(value);                         // добавление
minHeap.poll();                            // удаление вершины
minHeap.peek();                            // доступ к вершине

//Хеш таблица
import java.util.HashMap;
HashMap<KeyType, ValueType> hashTable = new HashMap<>();

// Операции:
hashTable.put(key, value);                  // вставка/обновление
hashTable.get(key);                        // поиск
hashTable.remove(key);                     // удаление
hashTable.size();                          // размер


